centreon/src/Core/Common/Infrastructure/Repository/FileDataStoreEngine.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Repository;
+
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Security\Encryption;
+
+class FileDataStoreEngine implements DataStorageEngineInterface
+{
+    private bool $inTransaction = false;
+
+    private string $workingDirectory = '';
+
+    private string $lastError = '';
+
+    /**
+     * @param string $absoluteMediaPath Absolute destination path for the media
+     * @param bool $throwsException Indicates whether an exception should be thrown in the event of an error,
+     *                              false by default
+     *
+     * @throws \Exception
+     */
+    public function __construct(private string $absoluteMediaPath, private bool $throwsException = false)
+    {
+        $this->absoluteMediaPath = realpath($absoluteMediaPath)
+            ?: throw new \Exception(sprintf('Path invalid \'%s\'', $absoluteMediaPath));
+    }
+
+    /**
+     * Defines whether an exception should be thrown in the event of an error.
+     *
+     * @param bool $throwsException
+     */
+    public function throwsException(bool $throwsException): void
+    {
+        $this->throwsException = $throwsException;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function rollbackTransaction(): bool
+    {
+        try {
+            $rollbackStatus = $this->rollbackFileSystem();
+            $this->inTransaction = false;
+
+            return $rollbackStatus;
+        } catch (\Throwable $ex) {
+            $this->lastError = $ex->getMessage();
+            if ($this->throwsException) {
+                throw $ex;
+            }
+
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function startTransaction(): bool
+    {
+        if ($this->inTransaction) {
+            return false;
+        }
+        $tempDirectory = str_replace(DIRECTORY_SEPARATOR, '-', Encryption::generateRandomString(32));
```

Comment:
`generateRandomString` is returning a base64 string, then you can explicitely tell `'/'`
```suggestion
        $tempDirectory = str_replace('/', '-', Encryption::generateRandomString(32));
```